### PR TITLE
ESP32-S3 and ESP-IDF don't play well with USB_CDC and need USB_SERIAL_JTAG

### PR DESCRIPTION
**Pull request by clydebarrow**
instead.

# What does this implement/fix?

Fixes compile errors when compiling for ESP-IDF on ESP32-S3 (and probably S2) due to change introduced with
#4853.

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

**Related issue or feature (if applicable):** fixes <link to issue>

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#<esphome-docs PR number goes here>

## Test Environment

- [ ] ESP32
- [x] ESP32 IDF
- [ ] ESP8266
- [ ] RP2040
- [ ] BK72xx
- [ ] RTL87xx

## Example entry for `config.yaml`:

```yaml
# Example config.yaml

```

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

File: esphome/components/logger/__init__.py [hunk 1]
```diff
@@ -171,8 +171,8 @@ def validate_local_no_higher_than_global(value):
                 CONF_HARDWARE_UART,
                 esp8266=UART0,
                 esp32=UART0,
-                esp32_s2=USB_CDC,
-                esp32_s3=USB_CDC,
+                esp32_s2=USB_SERIAL_JTAG,
+                esp32_s3=USB_SERIAL_JTAG,
```
Comment (kbx81) [inline esphome/components/logger/__init__.py]: In light of #5928, we can (and should) do this now:
```suggestion
                esp32_s2=USB_CDC,
                esp32_s3=USB_SERIAL_JTAG,
                esp32_s3_arduino=USB_CDC,
```
...the S2 does not support JTAG at all and CDC with the S3 won't work correctly with IDF.

**Review by kbx81 — DISMISSED** [review]

File: esphome/components/logger/__init__.py [hunk 2]
```diff
@@ -172,7 +172,8 @@ def validate_local_no_higher_than_global(value):
                 esp8266=UART0,
                 esp32=UART0,
                 esp32_s2=USB_CDC,
-                esp32_s3=USB_CDC,
+                esp32_s3=USB_SERIAL_JTAG,
```
Comment (kbx81) [inline esphome/components/logger/__init__.py]: One more time 😅 
```suggestion
                esp32_s3_idf=USB_SERIAL_JTAG,
```

**Review by kbx81 — DISMISSED** [review]

**Comment by clydebarrow** [comment]
gonna add the C3 as well.

**Comment by clydebarrow** [comment]
Tested on S3, C3 and plain ESP32. Also tested on RP2040.

**Comment by clydebarrow** [comment]
Arduino on ESP32-C3 is still not right. The current setup results in a default of UART0, which means no output when using the USB_CDC port. If I tweak the config to allow USB_CDC for C3 then it fails at compile time because `Serial` is not defined by Arduino - it's `Serial0`. If I replace `Serial` with `Serial0` then another compile error in the platformio main.cpp blocks it. If I work-around that I can get it to build and flash, but then it bootloops :-(

So for now I think it's best left as is and note that USB serial is not supported on the C3 under Arduino.

**Review by kbx81 — APPROVED** [review]

**Comment by kbx81** [comment]
> Arduino on ESP32-C3 is still not right. The current setup results in a default of UART0, which means no output when using the USB_CDC port. If I tweak the config to allow USB_CDC for C3 then it fails at compile time because `Serial` is not defined by Arduino - it's `Serial0`. If I replace `Serial` with `Serial0` then another compile error in the platformio main.cpp blocks it. If I work-around that I can get it to build and flash, but then it bootloops :-(

It behaves this way even [with this added](https://github.com/esphome/esphome/blob/0a188ad9d22a2b5a8060ec7ad3318cb9dba509e2/esphome/components/logger/__init__.py#L263-L265)? 🤔

**Comment by clydebarrow** [comment]
> It behaves this way even [with this added](https://github.com/esphome/esphome/blob/0a188ad9d22a2b5a8060ec7ad3318cb9dba509e2/esphome/components/logger/__init__.py#L263-L265)? 🤔

Yes. That works fine with S3, but on C3 `Serial` (without a 0) is undefined, not only in ESPHome code but also in platformio code. And fixing that leads to the bootloop. I just don't think the Arduino support for the C3 is mature.


```
/Users/clyde/.platformio/packages/framework-arduinoespressif32@3.20005.220925/cores/esp32/main.cpp: In function 'void app_main()':
/Users/clyde/.platformio/packages/framework-arduinoespressif32@3.20005.220925/cores/esp32/main.cpp:58:5: error: 'Serial' was not declared in this scope
     Serial.begin();
     ^~~~~~
```

**Comment by clydebarrow** [comment]
Oh, well this is weird, it's now working with Arduino on C3, with the default as UART0.

**Pull request merged**